src/context.ts

```diff
@@ -107,7 +107,6 @@ export async function runWithParentContext<R>(
 ): Promise<R> {
   return await asyncLocalCtx.run(
     {
-      ...pctx,
       workflowTimeoutMS: undefined, // Becomes deadline
       parentCtx: pctx,
       ...ctx,
```

Comment:
I *think* the ctx spread should to come before the explicit field settings. If ctx has a parentCtx or workflowTimeoutMS field (even if they are undefined) would overwrite the fields we are trying to set